packages/snaps-utils/src/manifest/validation.ts

```diff
@@ -141,7 +143,62 @@ export const SnapIdsStruct = refine(
 
 export type SnapIds = Infer<typeof SnapIdsStruct>;
 
-export const ChainIdsStruct = array(ChainIdStruct);
+export const ChainIdsStruct = refine(
+  size(array(ChainIdStruct), 1, Infinity),
+  'Chain IDs',
+  (value) => {
+    if (Array.isArray(value) && value.every((val) => isChainId(val))) {
+      return true;
+    }
+    return 'Expected caveat value to have type "string array"';
+  },
+);
+
+export const LookupMatchersStruct = refine(
+  object({
+    tlds: optional(size(array(string()), 1, Infinity)),
+    schemes: optional(size(array(string()), 1, Infinity)),
+  }),
+  'Lookup Matchers',
+  (value) => {
+    const satisfiesEmptyCondition =
```

Comment:
Why do we need special validation logic for this? It looks like the struct already would validate this.

If we need it, it should be tested.

Reply:
Similar thought process as above, but additionally i'm not sure the existing struct (using `optional`) would make sure that one or the other of the `matchers` properties could exist.

Reply:
I'm not sure I understand what you mean 🤔

Reply:
Since both properties are optional in `matchers`, I thought superstruct would allow an empty object.

Reply:
Ah you may be right, we could add a check for that. But everything else here seems unnecessary.

Reply:
Add a check for that where? Isn't that what we're doing here? If it's elsewhere then yeah it makes sense to remove this stuff but if I understand `refine` correctly, doesn't it override validation logic of the struct being passed in? If so, we would still need to make sure we throw an error in all the cases.

Reply:
Actually, I'll just union the types and not use `optional`

Reply:
Sounds good to me!

Reply:
Just pushed a change, it leaves the validation errors being a bit non-descriptive 🤷‍♂️